enhancements/machine-api/cluster-api-integration.md

```diff
@@ -0,0 +1,268 @@
+
+### Supported platforms
+
+The techincal preview aims to support: AWS, Azure, GCP.
+
+#### Cluster API resource management
+
+In order to maintain the lifecycle of Cluster API related resources, we will create a new operator `cluster-capi-operator`, this name was chosen for avoiding confusion with upstream Cluster API operator.
+This operator will be responsible for all administrative tasks related to the deployment of the Cluster API project within the cluster.
+During tech preview phase, the new operator will also manage all Cluster API related CRDs. All CRD manifests will placed in openshift forks of CAPI and will take from there with no changes.
+
+`cluster-capi-operator` and it's operands will be provisioned in a new `openshift-cluster-api` namespace.
+
+The operator will perform the following tasks:
+
+##### Reconcile FeatureGate object
+
+While Cluster API intergration is in tech preview, the operator will reconcile the cluster [`FeatureGate`](https://docs.openshift.com/container-platform/4.8/nodes/clusters/nodes-cluster-enabling-features.html) object and check for `ClusterAPIEnabled` feature gate presence.
+The operator will procceed with Cluster API installation if and only if the required feature gate is present.
+
+##### Deploy Cluster Machine Approver
+
+In order for Cluster API machines to succefully join the cluster, the Kubelet CSRs need to be approved.
+The operator will deploy a separate instance of the cluster-machine-approver, which will be configured to be used with Cluster API machines by providing [`--apigroup`](https://github.com/openshift/cluster-machine-approver/blob/master/main.go#L54) flag that was recently introduced.
+
+##### Install upstream CAPI operator
+
+We will use the upstream [Cluster API operator](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/proposals/20201020-capi-provider-operator.md) for managing CRDs and deploying cloud providers.
+`cluster-capi-operator` should first install the [upstream operator CRDs](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/proposals/20201020-capi-provider-operator.md#new-api-types) and then run the upstream operator itself using a `Deployment`.
+
+##### Deploy core Cluster API
+
+Once the upstream Cluster API Operator is installed, the next step is to create a `CoreProvider` CR along with a configmap that contains upstream Cluster API CRDs, Deployment, Webhooks and RBAC resources.
+The example usage is described [here](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/proposals/20201020-capi-provider-operator.md#air-gapped-environment).
+
+##### Deploy Cluster API infrastructure provider
+
+The `cluster-capi-operator` operator will create an appropriate `InfrastructureProvider` CR (based on the cluster platform) and a configmap that contains upstream Cluster API cloud provider CRDs, Deployment, Webhooks and RBAC resources.
```

Comment:
one question I have is many of the providers require a secret for the variables used in the manifests.
Do we need platform specific code to gather this and create the secret so the the infrastructure provider can
start correctly? Currently I am generating defaults only and creating those secrets, but the user would have to edit the secret if the defaults are not good enough.